### PR TITLE
Move ConvertArmConvolutionLayout functionality to ConvertSingleConvolutionToArm

### PR DESCRIPTION
**Pull request by mateusztabaka**

### CODE REVIEW

File: modules/arm_plugin/src/opset/pool_arm.cpp [hunk 1]
```diff
@@ -12,9 +12,9 @@ opset::v1::ArmMaxPool::ArmMaxPool(const ov::Output<Node>& arg,
                                   const ov::Shape& pads_begin,
                                   const ov::Shape& pads_end,
                                   const ov::Shape& kernel,
+                                  const ov::PartialShape& output_shape,
```
Comment (itikhono) [inline modules/arm_plugin/src/opset/pool_arm.cpp:15]: it's better to make it the last argument here and the files below

Comment (mateusztabaka) [reply modules/arm_plugin/src/opset/pool_arm.cpp:15]: all arguments after `output_shape` are optional. So if we want `output_shape` to not be optional, the last position to place it here is after `kernel` argument.

File: modules/arm_plugin/src/transformations/convert_pool_arm.cpp [hunk 2]
```diff
@@ -44,19 +54,35 @@ ArmPlugin::pass::ConvertArmMaxPoolV8::ConvertArmMaxPoolV8() {
...
+        if (rank < 4 || rank > 5) {
+            return false;
+        }
+        auto axis = max_pool->get_axis();
+        if (axis > 1 || (axis < 0 && axis > -static_cast<int64_t>(rank) - 1)) {
+            return false;
+        }
+        auto activations_transpose = transpose_on_input(max_pool->input_value(0), rank);
+        auto output_shape = transpose_output_shape(max_pool, rank);
+        auto arm_pool = std::make_shared<opset::v8::ArmMaxPool>(activations_transpose,
```
Comment (itikhono) [inline modules/arm_plugin/src/transformations/convert_pool_arm.cpp:73]: is ArmMaxPool class inherited from ov::op::v8::MaxPool?
If so, this conversion could potentially be called twice, I guess we need to check that this is not ArmMaxPool already.
```
auto arm_pool = std::dynamic_pointer_cast<ArmMaxPool>(m.get_match_root());
if (arm_pool)
   return false;
```

Comment (mateusztabaka) [reply modules/arm_plugin/src/transformations/convert_pool_arm.cpp:73]: done

**Review by itikhono — APPROVED** [review]

**Comment by mateusztabaka** [comment]
@allnes can you review?

**Review by allnes — APPROVED** [review]

**Pull request merged**